dev/tools/localization/gen_l10n.dart

```diff
@@ -302,184 +307,220 @@ bool _isValidGetterAndMethodName(String name) {
       );
-    if (pluralValueRE.hasMatch(bundle[key]))
-      classMethods.add(genPluralMethod(bundle, key));
-    else
-      classMethods.add(genSimpleMethod(bundle, key));
+    className = classNameString;
+  }
+
+  /// Scans [l10nDirectory] for arb files and parses them for language and locale
+  /// information.
+  void parseArbFiles() {
+    final List<FileSystemEntity> sortedFileSystemEntityList = l10nDirectory
+      .listSync()
+      .toList()
+      ..sort(sortFilesByPath);
+
+    for (FileSystemEntity entity in sortedFileSystemEntityList) {
+      final String entityPath = entity.path;
+      if (entity is File) {
+        final RegExp arbFilenameRE = RegExp(r'(\w+)\.arb$');
+        if (arbFilenameRE.hasMatch(entityPath)) {
+          final Map<String, dynamic> arbContents = json.decode(entity.readAsStringSync());
+          String localeString = arbContents['@@locale'];
+          if (localeString == null) {
+            final RegExp arbFilenameLocaleRE = RegExp(r'^[^_]*_(\w+)\.arb$');
+            final RegExpMatch arbFileMatch = arbFilenameLocaleRE.firstMatch(entityPath);
+            if (arbFileMatch == null) {
+              throw L10nException(
+                "The following .arb file's locale could not be determined: \n"
+                '$entityPath \n'
+                "Make sure that the locale is specified in the '@@locale' "
+                'property or as part of the filename (ie. file_en.arb)'
+              );
+            }
+
+            localeString = arbFilenameLocaleRE.firstMatch(entityPath)[1];
+          }
+
+          arbFilenames.add(entityPath);
+          final LocaleInfo localeInfo = LocaleInfo.fromString(localeString);
+          if (supportedLocales.contains(localeInfo))
+            throw L10nException(
+              'Multiple arb files with the same locale detected. \n'
+              'Ensure that there is exactly one arb file for each locale.'
+            );
+          supportedLocales.add(localeInfo);
+          supportedLanguageCodes.add('\'${localeInfo.languageCode}\'');
+        }
+      }
+    }
   }
 
-  outputFile.writeAsStringSync(
-    defaultFileTemplate
-      .replaceAll('@className', stringsClassName)
-      .replaceAll('@classMethods', classMethods.join('\n'))
-      .replaceAll('@importFile', _importFilePath(arbPathString, outputFileString))
-      .replaceAll('@supportedLocales', genSupportedLocaleProperty(supportedLocales))
-      .replaceAll('@supportedLanguageCodes', supportedLanguageCodes.toList().join(', '))
-  );
-
-  final ProcessResult pubGetResult = await Process.run('flutter', <String>['pub', 'get']);
-  if (pubGetResult.exitCode != 0) {
-    stderr.write(pubGetResult.stderr);
-    exit(1);
+  /// Generates the Dart class methods for the localizations class.
+  ///
+  /// The method parses [templateArbFile] and uses its resource ids as the
+  /// Dart method and getter names. It then uses its values to figure out
```

Comment:
"its values"?

Reply:
Updated to `It then uses each resource id's corresponding resource values to figure out how to define these getters.`